controllers/hazelcast/map_controller.go

```diff
@@ -0,0 +1,259 @@
+	if current.Name != last.Name {
+		return fmt.Errorf("name cannot be updated.")
+	}
+	if *current.BackupCount != *last.BackupCount {
+		return fmt.Errorf("backupCount cannot be updated.")
+	}
+	if *current.AsyncBackupCount != *last.AsyncBackupCount {
+		return fmt.Errorf("asyncBackupCount cannot be updated.")
+	}
+	if !util.IndexConfigSliceEquals(current.Indexes, last.Indexes) {
+		return fmt.Errorf("indexes cannot be updated.")
+	}
+	if current.PersistenceEnabled != last.PersistenceEnabled {
+		return fmt.Errorf("persistenceEnabled cannot be updated.")
+	}
+	if current.HazelcastResourceName != last.HazelcastResourceName {
+		return fmt.Errorf("hazelcastResourceName cannot be updated.")
+	}
+	return nil
+}
+
+func GetHazelcastClient(m *hazelcastv1alpha1.Map) (*hazelcast.Client, error) {
+	hzcl, ok := GetClient(types.NamespacedName{Name: m.Spec.HazelcastResourceName, Namespace: m.Namespace})
+	if !ok {
+		return nil, errors.NewInternalError(fmt.Errorf("Cannot connect to the cluster for %s", m.Spec.HazelcastResourceName))
+	}
+	if hzcl.client == nil || !hzcl.client.Running() {
+		return nil, fmt.Errorf("Trying to connect to the cluster %s", m.Spec.HazelcastResourceName)
+	}
+
+	return hzcl.client, nil
+}
+
+func (r *MapReconciler) ReconcileMapConfig(ctx context.Context, m *hazelcastv1alpha1.Map, cl *hazelcast.Client, createdBefore bool) (map[string]hazelcastv1alpha1.MapConfigState, error) {
+	ci := hazelcast.NewClientInternal(cl)
+	var req *proto.ClientMessage
+	if createdBefore {
+		req = codec.EncodeMCUpdateMapConfigRequest(
+			m.MapName(),
+			*m.Spec.TimeToLiveSeconds,
+			*m.Spec.MaxIdleSeconds,
+			hazelcastv1alpha1.EncodeEvictionPolicyType[m.Spec.Eviction.EvictionPolicy],
+			m.Spec.ReadBackupData,
+			*m.Spec.Eviction.MaxSize,
+			hazelcastv1alpha1.EncodeMaxSizePolicy[m.Spec.Eviction.MaxSizePolicy],
+		)
+	} else {
+		mapInput := codecTypes.DefaultAddMapConfigInput()
+		fillAddMapConfigInput(mapInput, m)
+		req = codec.EncodeDynamicConfigAddMapConfigRequest(mapInput)
+	}
+
+	memberStatuses := map[string]hazelcastv1alpha1.MapConfigState{}
+	var failedMembers strings.Builder
+	for _, member := range ci.OrderedMembers() {
+		if status, ok := m.Status.MemberStatuses[member.UUID.String()]; ok && status == hazelcastv1alpha1.MapSuccess {
+			memberStatuses[member.UUID.String()] = hazelcastv1alpha1.MapSuccess
+			continue
+		}
+		_, err := ci.InvokeOnMember(ctx, req, member.UUID, nil)
+		if err != nil {
+			memberStatuses[member.UUID.String()] = hazelcastv1alpha1.MapFailed
+			failedMembers.WriteString(member.UUID.String() + ", ")
+			continue
+		}
+		memberStatuses[member.UUID.String()] = hazelcastv1alpha1.MapSuccess
+	}
+	errString := failedMembers.String()
+	if errString == "" {
+		return memberStatuses, nil
+	}
+
+	return memberStatuses, fmt.Errorf("Error creating/updating the Map config %s for members %s.", m.MapName(), errString[:len(errString)-2])
```

Comment:
```suggestion
	if errString != "" {
		return memberStatuses, fmt.Errorf("Error creating/updating the Map config %s for members %s.", m.MapName(), errString[:len(errString)-2])
	}

	return memberStatuses, nil
```

This is very minor suggestion just to follow [the happy path to the left edge](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88) idiomatic rule